src/main/java/gtclassic/tile/GTTileChargeOMat.java

```diff
@@ -231,7 +231,7 @@ public void tryCharge() {
 		for (int i = 0; i < 9; ++i) {
 			if (!this.inventory.get(i).isEmpty()) {
 				if (hasEnergy()) {
```

Comment:
Should only iterate over slots if it has energy @e99999. (So if out of energy its should stop the loop.)

Reply:
I have the hasEnergy() called on 233 below it so full batteries would still get passed to the output reguardless of the internal energy status.